{"cells":[{"cell_type":"code","execution_count":null,"metadata":{"id":"Vv1FvfLq0d0R"},"outputs":[],"source":["#!/usr/bin/env python\n","# coding: utf-8\n","\n","# Biog5 migracion a colab\n","# Perez Ernesto Rafael rafaelperezctes@gmail.com\n","#Created on Mon Feb  3 14:02:59 2023\n","\n","#@author: Grupo Nº 1 LEMyP\n","\n","\"\"\"\n","Este codigo: Tarea: Migracion Biog5 a la nube\n","Son pruebas para instlacion de dependencias y programas necesarios del sistema de escritorio Biog5\n","\n","\"\"\""],"id":"Vv1FvfLq0d0R"},{"cell_type":"markdown","source":["# Ejecucion de otras celdas"],"metadata":{"id":"mamxFU23gzwD"},"id":"mamxFU23gzwD"},{"cell_type":"markdown","source":["1 Grabar bash e una celda"],"metadata":{"id":"BxEwbL98g-dc"},"id":"BxEwbL98g-dc"},{"cell_type":"code","source":["%%writefile 3.sh\n","ls -ln\n","echo \"Hola\""],"metadata":{"id":"-Cvt4OtzSVoK"},"id":"-Cvt4OtzSVoK","execution_count":null,"outputs":[]},{"cell_type":"code","source":["%%bash\n","sh 3.sh"],"metadata":{"id":"NmafDtxMSWUk"},"id":"NmafDtxMSWUk","execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["2. Grabar python de una celda"],"metadata":{"id":"_HmnDZc1hHm5"},"id":"_HmnDZc1hHm5"},{"cell_type":"code","source":["%%writefile 3.py\n","print('hola')\n","print('mundo')\n","print('soy rafael celda 4')"],"metadata":{"id":"gzxRacImSV6q"},"id":"gzxRacImSV6q","execution_count":null,"outputs":[]},{"cell_type":"code","source":["%run 3.py"],"metadata":{"id":"84V8AXyEhV83"},"id":"84V8AXyEhV83","execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["3. Ejecutar run process"],"metadata":{"id":"nuHQmR-ohpmY"},"id":"nuHQmR-ohpmY"},{"cell_type":"code","source":["import subprocess\n","from subprocess import run, PIPE\n","result = run(['sudo', 'apt-get install', 'ncbi-blast+', '&'], stdout=PIPE)\n","print(result)\n","result = run(['echo', 'Hola'], stdout=PIPE)\n","print(result)"],"metadata":{"id":"NJAudLhShomd"},"id":"NJAudLhShomd","execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"kOyrxd9E0d0c"},"source":["# 1) Instalación de librerias en google colab"],"id":"kOyrxd9E0d0c"},{"cell_type":"markdown","source":["Ideas\n","1-Puedo enlazar  a github y traer archivos con urlib\n","2-Si no existe se descargar agregar esa consistencia\n","\n","\n"],"metadata":{"id":"1jXXTOH1WlOz"},"id":"1jXXTOH1WlOz"},{"cell_type":"markdown","source":["1.1) pip"],"metadata":{"id":"YrOUrnsmAo3P"},"id":"YrOUrnsmAo3P"},{"cell_type":"code","source":["# librerias biog5\n","!pip install biopython scipy numpy pandas matplotlib nglview seaborn\n","!sudo apt-get update\n","!sudo apt-get install -y ncbi-blast+\n","!sudo apt-get install -y clustalo\n","#ncbi-blast+ (2.9.0-2)\n","#!pip install biopython==1.81 scipy==1.10.1 numpy==1.22.4 pandas==1.4.4 matplotlib==3.7.1 nglview==3.0.3"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"ibgAGRNHxwHp","executionInfo":{"status":"ok","timestamp":1679341481440,"user_tz":180,"elapsed":44070,"user":{"displayName":"rafael perez","userId":"12990849696245978067"}},"outputId":"a8836658-9509-4bd1-b857-4938062aa2a2"},"id":"ibgAGRNHxwHp","execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/\n","Collecting biopython\n","  Downloading biopython-1.81-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.1 MB)\n","\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m3.1/3.1 MB\u001b[0m \u001b[31m24.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hRequirement already satisfied: scipy in /usr/local/lib/python3.9/dist-packages (1.10.1)\n","Requirement already satisfied: numpy in /usr/local/lib/python3.9/dist-packages (1.22.4)\n","Requirement already satisfied: pandas in /usr/local/lib/python3.9/dist-packages (1.4.4)\n","Requirement already satisfied: matplotlib in /usr/local/lib/python3.9/dist-packages (3.7.1)\n","Collecting nglview\n","  Downloading nglview-3.0.3.tar.gz (5.7 MB)\n","\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m5.7/5.7 MB\u001b[0m \u001b[31m48.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25h  Installing build dependencies ... \u001b[?25l\u001b[?25hdone\n","  Getting requirements to build wheel ... \u001b[?25l\u001b[?25hdone\n","  Preparing metadata (pyproject.toml) ... \u001b[?25l\u001b[?25hdone\n","Requirement already satisfied: seaborn in /usr/local/lib/python3.9/dist-packages (0.12.2)\n","Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.9/dist-packages (from pandas) (2022.7.1)\n","Requirement already satisfied: python-dateutil>=2.8.1 in /usr/local/lib/python3.9/dist-packages (from pandas) (2.8.2)\n","Requirement already satisfied: fonttools>=4.22.0 in /usr/local/lib/python3.9/dist-packages (from matplotlib) (4.39.0)\n","Requirement already satisfied: pyparsing>=2.3.1 in /usr/local/lib/python3.9/dist-packages (from matplotlib) (3.0.9)\n","Requirement already satisfied: kiwisolver>=1.0.1 in /usr/local/lib/python3.9/dist-packages (from matplotlib) (1.4.4)\n","Requirement already satisfied: pillow>=6.2.0 in /usr/local/lib/python3.9/dist-packages (from matplotlib) (8.4.0)\n","Requirement already satisfied: contourpy>=1.0.1 in /usr/local/lib/python3.9/dist-packages (from matplotlib) (1.0.7)\n","Requirement already satisfied: packaging>=20.0 in /usr/local/lib/python3.9/dist-packages (from matplotlib) (23.0)\n","Requirement already satisfied: importlib-resources>=3.2.0 in /usr/local/lib/python3.9/dist-packages (from matplotlib) (5.12.0)\n","Requirement already satisfied: cycler>=0.10 in /usr/local/lib/python3.9/dist-packages (from matplotlib) (0.11.0)\n","Requirement already satisfied: jupyterlab-widgets in /usr/local/lib/python3.9/dist-packages (from nglview) (3.0.5)\n","Requirement already satisfied: ipywidgets>=7 in /usr/local/lib/python3.9/dist-packages (from nglview) (7.7.1)\n","Requirement already satisfied: zipp>=3.1.0 in /usr/local/lib/python3.9/dist-packages (from importlib-resources>=3.2.0->matplotlib) (3.15.0)\n","Requirement already satisfied: widgetsnbextension~=3.6.0 in /usr/local/lib/python3.9/dist-packages (from ipywidgets>=7->nglview) (3.6.2)\n","Requirement already satisfied: traitlets>=4.3.1 in /usr/local/lib/python3.9/dist-packages (from ipywidgets>=7->nglview) (5.7.1)\n","Requirement already satisfied: ipython-genutils~=0.2.0 in /usr/local/lib/python3.9/dist-packages (from ipywidgets>=7->nglview) (0.2.0)\n","Requirement already satisfied: ipykernel>=4.5.1 in /usr/local/lib/python3.9/dist-packages (from ipywidgets>=7->nglview) (5.3.4)\n","Requirement already satisfied: ipython>=4.0.0 in /usr/local/lib/python3.9/dist-packages (from ipywidgets>=7->nglview) (7.9.0)\n","Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.9/dist-packages (from python-dateutil>=2.8.1->pandas) (1.15.0)\n","Requirement already satisfied: tornado>=4.2 in /usr/local/lib/python3.9/dist-packages (from ipykernel>=4.5.1->ipywidgets>=7->nglview) (6.2)\n","Requirement already satisfied: jupyter-client in /usr/local/lib/python3.9/dist-packages (from ipykernel>=4.5.1->ipywidgets>=7->nglview) (6.1.12)\n","Collecting jedi>=0.10\n","  Downloading jedi-0.18.2-py2.py3-none-any.whl (1.6 MB)\n","\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m1.6/1.6 MB\u001b[0m \u001b[31m45.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hRequirement already satisfied: pygments in /usr/local/lib/python3.9/dist-packages (from ipython>=4.0.0->ipywidgets>=7->nglview) (2.6.1)\n","Requirement already satisfied: setuptools>=18.5 in /usr/local/lib/python3.9/dist-packages (from ipython>=4.0.0->ipywidgets>=7->nglview) (63.4.3)\n","Requirement already satisfied: backcall in /usr/local/lib/python3.9/dist-packages (from ipython>=4.0.0->ipywidgets>=7->nglview) (0.2.0)\n","Requirement already satisfied: pexpect in /usr/local/lib/python3.9/dist-packages (from ipython>=4.0.0->ipywidgets>=7->nglview) (4.8.0)\n","Requirement already satisfied: pickleshare in /usr/local/lib/python3.9/dist-packages (from ipython>=4.0.0->ipywidgets>=7->nglview) (0.7.5)\n","Requirement already satisfied: decorator in /usr/local/lib/python3.9/dist-packages (from ipython>=4.0.0->ipywidgets>=7->nglview) (4.4.2)\n","Requirement already satisfied: prompt-toolkit<2.1.0,>=2.0.0 in /usr/local/lib/python3.9/dist-packages (from ipython>=4.0.0->ipywidgets>=7->nglview) (2.0.10)\n","Requirement already satisfied: notebook>=4.4.1 in /usr/local/lib/python3.9/dist-packages (from widgetsnbextension~=3.6.0->ipywidgets>=7->nglview) (6.5.3)\n","Requirement already satisfied: parso<0.9.0,>=0.8.0 in /usr/local/lib/python3.9/dist-packages (from jedi>=0.10->ipython>=4.0.0->ipywidgets>=7->nglview) (0.8.3)\n","Requirement already satisfied: jupyter-core>=4.6.1 in /usr/local/lib/python3.9/dist-packages (from notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets>=7->nglview) (5.3.0)\n","Requirement already satisfied: argon2-cffi in /usr/local/lib/python3.9/dist-packages (from notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets>=7->nglview) (21.3.0)\n","Requirement already satisfied: nbconvert>=5 in /usr/local/lib/python3.9/dist-packages (from notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets>=7->nglview) (6.5.4)\n","Requirement already satisfied: nbclassic>=0.4.7 in /usr/local/lib/python3.9/dist-packages (from notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets>=7->nglview) (0.5.3)\n","Requirement already satisfied: jinja2 in /usr/local/lib/python3.9/dist-packages (from notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets>=7->nglview) (3.1.2)\n","Requirement already satisfied: terminado>=0.8.3 in /usr/local/lib/python3.9/dist-packages (from notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets>=7->nglview) (0.17.1)\n","Requirement already satisfied: prometheus-client in /usr/local/lib/python3.9/dist-packages (from notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets>=7->nglview) (0.16.0)\n","Requirement already satisfied: nest-asyncio>=1.5 in /usr/local/lib/python3.9/dist-packages (from notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets>=7->nglview) (1.5.6)\n","Requirement already satisfied: nbformat in /usr/local/lib/python3.9/dist-packages (from notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets>=7->nglview) (5.7.3)\n","Requirement already satisfied: Send2Trash>=1.8.0 in /usr/local/lib/python3.9/dist-packages (from notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets>=7->nglview) (1.8.0)\n","Requirement already satisfied: pyzmq>=17 in /usr/local/lib/python3.9/dist-packages (from notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets>=7->nglview) (23.2.1)\n","Requirement already satisfied: wcwidth in /usr/local/lib/python3.9/dist-packages (from prompt-toolkit<2.1.0,>=2.0.0->ipython>=4.0.0->ipywidgets>=7->nglview) (0.2.6)\n","Requirement already satisfied: ptyprocess>=0.5 in /usr/local/lib/python3.9/dist-packages (from pexpect->ipython>=4.0.0->ipywidgets>=7->nglview) (0.7.0)\n","Requirement already satisfied: platformdirs>=2.5 in /usr/local/lib/python3.9/dist-packages (from jupyter-core>=4.6.1->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets>=7->nglview) (3.1.1)\n","Requirement already satisfied: jupyter-server>=1.8 in /usr/local/lib/python3.9/dist-packages (from nbclassic>=0.4.7->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets>=7->nglview) (1.23.6)\n","Requirement already satisfied: notebook-shim>=0.1.0 in /usr/local/lib/python3.9/dist-packages (from nbclassic>=0.4.7->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets>=7->nglview) (0.2.2)\n","Requirement already satisfied: lxml in /usr/local/lib/python3.9/dist-packages (from nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets>=7->nglview) (4.9.2)\n","Requirement already satisfied: defusedxml in /usr/local/lib/python3.9/dist-packages (from nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets>=7->nglview) (0.7.1)\n","Requirement already satisfied: bleach in /usr/local/lib/python3.9/dist-packages (from nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets>=7->nglview) (6.0.0)\n","Requirement already satisfied: mistune<2,>=0.8.1 in /usr/local/lib/python3.9/dist-packages (from nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets>=7->nglview) (0.8.4)\n","Requirement already satisfied: pandocfilters>=1.4.1 in /usr/local/lib/python3.9/dist-packages (from nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets>=7->nglview) (1.5.0)\n","Requirement already satisfied: entrypoints>=0.2.2 in /usr/local/lib/python3.9/dist-packages (from nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets>=7->nglview) (0.4)\n","Requirement already satisfied: tinycss2 in /usr/local/lib/python3.9/dist-packages (from nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets>=7->nglview) (1.2.1)\n","Requirement already satisfied: nbclient>=0.5.0 in /usr/local/lib/python3.9/dist-packages (from nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets>=7->nglview) (0.7.2)\n","Requirement already satisfied: MarkupSafe>=2.0 in /usr/local/lib/python3.9/dist-packages (from nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets>=7->nglview) (2.1.2)\n","Requirement already satisfied: beautifulsoup4 in /usr/local/lib/python3.9/dist-packages (from nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets>=7->nglview) (4.11.2)\n","Requirement already satisfied: jupyterlab-pygments in /usr/local/lib/python3.9/dist-packages (from nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets>=7->nglview) (0.2.2)\n","Requirement already satisfied: jsonschema>=2.6 in /usr/local/lib/python3.9/dist-packages (from nbformat->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets>=7->nglview) (4.3.3)\n","Requirement already satisfied: fastjsonschema in /usr/local/lib/python3.9/dist-packages (from nbformat->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets>=7->nglview) (2.16.3)\n","Requirement already satisfied: argon2-cffi-bindings in /usr/local/lib/python3.9/dist-packages (from argon2-cffi->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets>=7->nglview) (21.2.0)\n","Requirement already satisfied: attrs>=17.4.0 in /usr/local/lib/python3.9/dist-packages (from jsonschema>=2.6->nbformat->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets>=7->nglview) (22.2.0)\n","Requirement already satisfied: pyrsistent!=0.17.0,!=0.17.1,!=0.17.2,>=0.14.0 in /usr/local/lib/python3.9/dist-packages (from jsonschema>=2.6->nbformat->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets>=7->nglview) (0.19.3)\n","Requirement already satisfied: websocket-client in /usr/local/lib/python3.9/dist-packages (from jupyter-server>=1.8->nbclassic>=0.4.7->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets>=7->nglview) (1.5.1)\n","Requirement already satisfied: anyio<4,>=3.1.0 in /usr/local/lib/python3.9/dist-packages (from jupyter-server>=1.8->nbclassic>=0.4.7->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets>=7->nglview) (3.6.2)\n","Requirement already satisfied: cffi>=1.0.1 in /usr/local/lib/python3.9/dist-packages (from argon2-cffi-bindings->argon2-cffi->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets>=7->nglview) (1.15.1)\n","Requirement already satisfied: soupsieve>1.2 in /usr/local/lib/python3.9/dist-packages (from beautifulsoup4->nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets>=7->nglview) (2.4)\n","Requirement already satisfied: webencodings in /usr/local/lib/python3.9/dist-packages (from bleach->nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets>=7->nglview) (0.5.1)\n","Requirement already satisfied: sniffio>=1.1 in /usr/local/lib/python3.9/dist-packages (from anyio<4,>=3.1.0->jupyter-server>=1.8->nbclassic>=0.4.7->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets>=7->nglview) (1.3.0)\n","Requirement already satisfied: idna>=2.8 in /usr/local/lib/python3.9/dist-packages (from anyio<4,>=3.1.0->jupyter-server>=1.8->nbclassic>=0.4.7->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets>=7->nglview) (3.4)\n","Requirement already satisfied: pycparser in /usr/local/lib/python3.9/dist-packages (from cffi>=1.0.1->argon2-cffi-bindings->argon2-cffi->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets>=7->nglview) (2.21)\n","Building wheels for collected packages: nglview\n","  Building wheel for nglview (pyproject.toml) ... \u001b[?25l\u001b[?25hdone\n","  Created wheel for nglview: filename=nglview-3.0.3-py3-none-any.whl size=8057550 sha256=92a2ad68797d2d51cda89fa89e56bd8c3f08c162396f19fcfacd7b5f282bb113\n","  Stored in directory: /root/.cache/pip/wheels/ed/12/ba/4e227d89934c5d7bdf91387286e45dc868613ab32fa6ce36cf\n","Successfully built nglview\n","Installing collected packages: jedi, biopython, nglview\n","Successfully installed biopython-1.81 jedi-0.18.2 nglview-3.0.3\n","Reading package lists... Done\n","Building dependency tree       \n","Reading state information... Done\n","The following additional packages will be installed:\n","  ncbi-data\n","The following NEW packages will be installed:\n","  ncbi-blast+ ncbi-data\n","0 upgraded, 2 newly installed, 0 to remove and 23 not upgraded.\n","Need to get 14.6 MB of archives.\n","After this operation, 74.2 MB of additional disk space will be used.\n","Get:1 http://archive.ubuntu.com/ubuntu focal/universe amd64 ncbi-data all 6.1.20170106+dfsg1-8 [3,518 kB]\n","Get:2 http://archive.ubuntu.com/ubuntu focal/universe amd64 ncbi-blast+ amd64 2.9.0-2 [11.1 MB]\n","Fetched 14.6 MB in 0s (29.4 MB/s)\n","Selecting previously unselected package ncbi-data.\n","(Reading database ... 128276 files and directories currently installed.)\n","Preparing to unpack .../ncbi-data_6.1.20170106+dfsg1-8_all.deb ...\n","Unpacking ncbi-data (6.1.20170106+dfsg1-8) ...\n","Selecting previously unselected package ncbi-blast+.\n","Preparing to unpack .../ncbi-blast+_2.9.0-2_amd64.deb ...\n","Unpacking ncbi-blast+ (2.9.0-2) ...\n","Setting up ncbi-data (6.1.20170106+dfsg1-8) ...\n","Setting up ncbi-blast+ (2.9.0-2) ...\n","Processing triggers for man-db (2.9.1-1) ...\n","Processing triggers for hicolor-icon-theme (0.17-2) ...\n"]}]},{"cell_type":"code","source":["import subprocess\n","from subprocess import run, PIPE\n","result = run(['blastp', '-h'], stdout=PIPE)\n","print(result)\n","!blastp -version"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"--jZNv8ok_5S","executionInfo":{"status":"ok","timestamp":1679341484473,"user_tz":180,"elapsed":11,"user":{"displayName":"rafael perez","userId":"12990849696245978067"}},"outputId":"4e836914-691b-4ecc-ae9b-860f7ae0cb33"},"id":"--jZNv8ok_5S","execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["CompletedProcess(args=['blastp', '-h'], returncode=0, stdout=b\"USAGE\\n  blastp [-h] [-help] [-import_search_strategy filename]\\n    [-export_search_strategy filename] [-task task_name] [-db database_name]\\n    [-dbsize num_letters] [-gilist filename] [-seqidlist filename]\\n    [-negative_gilist filename] [-negative_seqidlist filename]\\n    [-taxids taxids] [-negative_taxids taxids] [-taxidlist filename]\\n    [-negative_taxidlist filename] [-ipglist filename]\\n    [-negative_ipglist filename] [-entrez_query entrez_query]\\n    [-db_soft_mask filtering_algorithm] [-db_hard_mask filtering_algorithm]\\n    [-subject subject_input_file] [-subject_loc range] [-query input_file]\\n    [-out output_file] [-evalue evalue] [-word_size int_value]\\n    [-gapopen open_penalty] [-gapextend extend_penalty]\\n    [-qcov_hsp_perc float_value] [-max_hsps int_value]\\n    [-xdrop_ungap float_value] [-xdrop_gap float_value]\\n    [-xdrop_gap_final float_value] [-searchsp int_value] [-seg SEG_options]\\n    [-soft_masking soft_masking] [-matrix matrix_name]\\n    [-threshold float_value] [-culling_limit int_value]\\n    [-best_hit_overhang float_value] [-best_hit_score_edge float_value]\\n    [-subject_besthit] [-window_size int_value] [-lcase_masking]\\n    [-query_loc range] [-parse_deflines] [-outfmt format] [-show_gis]\\n    [-num_descriptions int_value] [-num_alignments int_value]\\n    [-line_length line_length] [-html] [-sorthits sort_hits]\\n    [-sorthsps sort_hsps] [-max_target_seqs num_sequences]\\n    [-num_threads int_value] [-ungapped] [-remote] [-comp_based_stats compo]\\n    [-use_sw_tback] [-version]\\n\\nDESCRIPTION\\n   Protein-Protein BLAST 2.9.0+\\n\\nUse '-help' to print detailed descriptions of command line arguments\\n\")\n","blastp: 2.9.0+\n"," Package: blast 2.9.0, build Sep 30 2019 01:57:31\n"]}]},{"cell_type":"code","source":["#clono biog5\n","!rm -rf HPV-notebook\n","!git clone https://github.com/biog5/HPV-notebook.git\n","# seteo path de la ruta\n","#!cd /content/HPV-notebook/\n","import os\n","local_path = '/content/HPV-notebook/'\n","os.chdir(local_path)\n","!ls"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"acrqQs1KTjlw","executionInfo":{"status":"ok","timestamp":1679341494637,"user_tz":180,"elapsed":5003,"user":{"displayName":"rafael perez","userId":"12990849696245978067"}},"outputId":"812d7cbd-909b-47d9-b2f7-e1641ea59347"},"id":"acrqQs1KTjlw","execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["Cloning into 'HPV-notebook'...\n","remote: Enumerating objects: 1007, done.\u001b[K\n","remote: Counting objects: 100% (1007/1007), done.\u001b[K\n","remote: Compressing objects: 100% (808/808), done.\u001b[K\n","remote: Total 1007 (delta 202), reused 993 (delta 192), pack-reused 0\u001b[K\n","Receiving objects: 100% (1007/1007), 16.92 MiB | 6.99 MiB/s, done.\n","Resolving deltas: 100% (202/202), done.\n","Updating files: 100% (1549/1549), done.\n"," Alineamientos0.py\t\t      Main.py\n"," AlineamientosMSA.py\t\t      ModelosHMM.py\n"," Alineamientos.py\t\t      pipeline_biog5.sh\n"," ApiBlast.py\t\t\t      Propuesta_de_Trabajo_Final_Grupo5_v1.pdf\n"," BD\t\t\t\t      Publicaciones\n","'BD - copia'\t\t\t      __pycache__\n"," blast.xml\t\t\t      README.md\n"," CargaBD.py\t\t\t      requerimientos.txt\n"," clustal-omega-1.2.2-win64\t      temp\n"," ConfigurarRiesgos.py\t\t      TFBiog5_v1.0.pdf\n"," EstructurasNCBI.py\t\t      Utils.py\n"," EstructurasNCBI-visualizador.ipynb   VariantesGenomas.py\n"," IMG\t\t\t\t      VariantesProteinas.py\n"," LICENSE\t\t\t      version.txt\n"," ListaObjetivos.txt\n"]}]},{"cell_type":"code","source":["# Clono rama\n","!rm -rf HPV-notebook\n","!git clone -b develop https://github.com/biog5/HPV-notebook.git\n","# seteo path de la ruta\n","#!cd /content/HPV-notebook/\n","import os\n","local_path = '/content/HPV-notebook/'\n","os.chdir(local_path)\n","!ls"],"metadata":{"id":"ZxZEOfcgQ8zK"},"id":"ZxZEOfcgQ8zK","execution_count":null,"outputs":[]},{"cell_type":"code","source":["from google.colab import files\n","files.upload() # eligo lo editado"],"metadata":{"id":"Q_kaW11FzCJQ"},"id":"Q_kaW11FzCJQ","execution_count":null,"outputs":[]},{"cell_type":"code","source":["#os.path.exists('BD/HPV16E1_high_vs_E1_high.blast')\n","os.path.exists('BD/low_riskE1.fasta')\n","#os.listdir('BD/')\n","#import subprocess\n","#from subprocess import run, PIPE\n","#blast1= ['blastp', '-query', 'BD/HPV16E1.fasta', '-subject', 'BD/low_riskE1.fasta', '-out', 'BD/HPV16E1_high_vs_E1_low.blast']\n","#subprocess.run(blast1,stdout=PIPE)"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"16nR6v9uez6_","executionInfo":{"status":"ok","timestamp":1679341516647,"user_tz":180,"elapsed":277,"user":{"displayName":"rafael perez","userId":"12990849696245978067"}},"outputId":"294ce506-4ddf-4f6b-8cab-e77f8afe843c"},"id":"16nR6v9uez6_","execution_count":null,"outputs":[{"output_type":"execute_result","data":{"text/plain":["True"]},"metadata":{},"execution_count":5}]},{"cell_type":"code","source":["import Main\n","Main.Menu()"],"metadata":{"id":"0UmCQyP7l-1N"},"id":"0UmCQyP7l-1N","execution_count":null,"outputs":[]},{"cell_type":"code","source":["#!/usr/bin/env python\n","# coding: utf-8\n","\n","# # TP6 Grupo Nº 5\n","# Perez Ernesto Rafael rafaelperezctes@gmail.com Sofia Erdozain sofierdozain@gmail.com\n","# Created on Wed Nov  3 14:02:59 2021\n","\n","# @author: Grupo Nº 5\n","\n","import sys\n","\n","\n","def Menu():\n","    print(\"#####################################\")\n","    print(\"###### Bienvenidos a BIOG5 ######\")\n","    print(\"#####################################\")\n","    print()\n","    print(\"Que desea realizar:\")\n","    print(\"1- Alineamientos de a pares usando Blast\")\n","    print(\"2- Alineamientos multiples usando Clustal Omega\")\n","    print(\"3- Modelos HMM\")\n","    print(\"4- Analisis de variantes en proteinas\")\n","    print(\"5- Analisis de variantes en genomas\")\n","    print(\"6- Estructuras NCBI\")\n","    print(\"7- Configuar Riesgos\")\n","    print(\"8- Salir\")\n","    opcion_principal = int(input(\"Ingrese una opción: \"))\n","\n","    if opcion_principal == 1:\n","        import Alineamientos\n","        Alineamientos.Menu()\n","\n","    if opcion_principal == 2:\n","        import AlineamientosMSA\n","        AlineamientosMSA.Menu()\n","\n","    if opcion_principal == 3:\n","        import ModelosHMM\n","        ModelosHMM.Menu()\n","\n","    if opcion_principal == 4:\n","        import VariantesProteinas\n","        VariantesProteinas.Menu()\n","\n","    if opcion_principal == 5:\n","        import VariantesGenomas\n","        VariantesGenomas.Menu()\n","\n","    if opcion_principal == 6:\n","        import EstructurasNCBI\n","        EstructurasNCBI.Menu()\n","\n","    if opcion_principal == 7:\n","        import ConfigurarRiesgos\n","        ConfigurarRiesgos.Menu()\n","\n","    if opcion_principal == 8:\n","        print(\"Gracias por utilizar BIOG5\")\n","        sys.exit()\n","\n","\n","if __name__ == '__main__':\n","    Menu()\n"],"metadata":{"id":"hcl_rDwZrQIR"},"id":"hcl_rDwZrQIR","execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["# Pruebas extras"],"metadata":{"id":"QJZiT7-MmfHi"},"id":"QJZiT7-MmfHi"},{"cell_type":"code","source":["#@title Texto de título predeterminado\n","variable_name = \" #@param {type:\"slider\", min:0, max:100, step:1}\n","variable_name = \"\" #@param {type:\"raw\"}\n"],"metadata":{"id":"F74VjgG2JFuj"},"id":"F74VjgG2JFuj","execution_count":null,"outputs":[]},{"cell_type":"code","source":["#!ls\n","#!blastp -h\n","import subprocess\n","from subprocess import run, PIPE, Popen\n","#result = subprocess.run('blastp -query BD/HPV16E1.fasta -subject BD/low_riskE1.fasta -out BD/HPV16E1_high_vs_E1_low.blast')\n","result = run(['blastp', '-query', 'BD/HPV16E1.fasta', '-subject', 'BD/low_riskE1.fasta', '-out', 'BD/HPV16E1_high_vs_E1_low.blast'], stdout=PIPE)\n","print(result)"],"metadata":{"id":"iO_KeUuMtYJB"},"id":"iO_KeUuMtYJB","execution_count":null,"outputs":[]},{"cell_type":"code","source":["!pip install --upgrade google-colab\n","\n","from google.colab import kernel\n","kernel.invoke_as_main()\n","otra_celda = \"\"\"\n","#@title Esta es la celda que quiero llamar\n","print(\"Hola desde la otra celda\")\n","\"\"\"\n","#kernel.get_instance().run_cell(otra_celda)\n"],"metadata":{"id":"NM4BYag7uL3L"},"id":"NM4BYag7uL3L","execution_count":null,"outputs":[]},{"cell_type":"code","source":["print(\"celda 10\")\n","cell = get_ipython().run_cell_magic('javascript', '', 'IPython.notebook.get_cell(2)')"],"metadata":{"id":"A2DP5aqOuMHi"},"id":"A2DP5aqOuMHi","execution_count":null,"outputs":[]},{"cell_type":"code","source":["get_ipython().run_cell(cell)"],"metadata":{"id":"LSIE-kfPGDtg"},"id":"LSIE-kfPGDtg","execution_count":null,"outputs":[]},{"cell_type":"code","source":["##print(\"celda 11\")\n","In[54]\n","#In[43]\n","#get_ipython().run_cell(print(\"celda 10\"))"],"metadata":{"id":"F3vdvFsJEl5Y"},"id":"F3vdvFsJEl5Y","execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["# aliniamientos"],"metadata":{"id":"j8HnFammDafy"},"id":"j8HnFammDafy"},{"cell_type":"code","source":["import requests\n","from io import StringIO\n","from Bio.Blast import NCBIXML\n","\n","import requests\n","from Bio.Blast import NCBIXML\n","\n","def run_blast(sequence, program='blastp', database='nr', email='your_email@example.com'):\n","    base_url = 'https://blast.ncbi.nlm.nih.gov/Blast.cgi'\n","    params = {\n","        'CMD': 'Put',\n","        'PROGRAM': program,\n","        'DATABASE': database,\n","        'QUERY': sequence,\n","    }\n","    response = requests.post(base_url, params=params)\n","    rid = response.text.split('&')[0].split('=')[1]\n","    \n","    params = {\n","        'CMD': 'Get',\n","        'FORMAT_TYPE': 'XML',\n","        'RID': rid,\n","    }\n","    \n","    while True:\n","        response = requests.get(base_url, params=params)\n","        if response.status_code == 200:\n","            break\n","    \n","    return NCBIXML.read(StringIO(response.text))\n","\n","\n","\n","def AlinearRiesgos_API():\n","    global cont_querry\n","    cont_querry = 0\n","    for gen in genes:\n","        archivo_low = \"BD/low_risk\" + gen + '.fasta'\n","        archivo_unspecified = \"BD/unspecified_risk\" + gen + '.fasta'\n","        archivo_high2 = \"BD/high_risk\" + gen + '.fasta'\n","\n","        for genoma in high_risk:  # por cada genoma y gen de alto riesgo\n","            archivo_entrada = BuscarNombres(str(genoma), str(gen))\n","            if archivo_entrada != None:\n","                archivo_high = 'BD/' + archivo_entrada + '.fasta'\n","                archivo_salida = \"BD/\" + genoma + gen + \"_high_vs_\" + gen + \"_low.blast\"\n","                archivo_salida2 = \"BD/\" + genoma + gen + \"_high_vs_\" + gen + \"_unspecified.blast\"\n","                archivo_salida3 = \"BD/\" + genoma + gen + \"_high_vs_\" + gen + \"_high.blast\"\n","\n","                # Ejecutar BLAST a través de la API de NCBI\n","                result = run_blast(open(archivo_high).read(), program='blastp', database='nr')\n","                print(\"Corriendo BLASTP ---> Cepa:\", genoma, \" Proteina:\", gen, \" Contra grupo: Bajo riego \")\n","                result2 = run_blast(open(archivo_high).read(), program='blastp', database='nr')\n","                print(\"Corriendo BLASTP ---> Cepa:\", genoma, \" Proteina:\", gen, \" Contra grupo: No especificado riego \")\n","                result3 = run_blast(open(archivo_high).read(), program='blastp', database='nr')\n","                print(\"Corriendo BLASTP ---> Cepa:\", genoma, \" Proteina:\", gen, \" Contra grupo: Alto riego \")\n","\n","                # Guardar resultados en archivos de salida\n","                with open(archivo_salida, 'w') as f:\n","                    f.write(result)\n","                with open(archivo_salida2, 'w') as f:\n","                    f.write(result2)\n","                with open(archivo_salida3, 'w') as f:\n","                    f.write(result3)\n","\n","                archivos_temp.append(archivo_salida)\n","                archivos_temp.append(archivo_salida2)\n","                archivos_temp.append(archivo_salida3)\n","\n","                # Parsear archivos de salida y almacenar resultados en estructura de datos\n","                resultados = NCBIXML.parse(StringIO(result))\n","                resultados2 = NCBIXML.parse(StringIO(result2))\n","                resultados3 = NCBIXML.parse(StringIO(result3))\n","\n","                for r in resultados:\n","                    for h in r.alignments:\n","                        for hsp in h.hsps:\n","                            resultados_high_vs_low[genoma][gen]['high'].append((h.hit_def, hsp.score))\n","                for r in resultados2:\n","                    for h in r.alignments:\n","                        for hsp in h.hsps:\n","                            resultados_high_vs_unspecified[genoma][gen]['high'].append((h.hit_def, hsp.score))\n","                for r in resultados3:\n","                    for h in r.alignments:\n","                        for hsp in h.hsps:\n","                            resultados_high_vs_high[genoma][gen]['high'].append((h.hit_def, hsp.score))"],"metadata":{"id":"WjB6sFVMGdkC"},"id":"WjB6sFVMGdkC","execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["# graba .blast"],"metadata":{"id":"LPSu9eR5J4pA"},"id":"LPSu9eR5J4pA"},{"cell_type":"code","source":["import requests\n","import time\n","import os\n","\n","def run_blast(query_file, subject_file, output_file):\n","    base_url = 'https://blast.ncbi.nlm.nih.gov/Blast.cgi'\n","    \n","    # Leer el contenido de los archivos\n","    with open(query_file, 'r') as f:\n","        query_sequence = f.read()\n","    with open(subject_file, 'r') as f:\n","        subject_sequence = f.read()\n","\n","    # Configurar los parámetros para la solicitud POST\n","    payload = {\n","        'CMD': 'Put',\n","        'PROGRAM': 'blastp',  # Asegúrate de usar 'blastn' para ADN\n","        'QUERY': query_sequence,\n","        'SUBJECTS': subject_sequence\n","    }\n","    \n","    # Enviar la solicitud POST\n","    response = requests.post(base_url, data=payload)\n","    response.raise_for_status()\n","    \n","    # Extraer el ID de la búsqueda (RID) de la respuesta\n","    result_page = response.text\n","    rid = None\n","    for line in result_page.splitlines():\n","        if line.startswith(\"RID = \"):\n","            rid = line[6:]\n","    \n","    if rid is None:\n","        raise Exception(\"No se pudo obtener el RID del resultado\")\n","\n","    # Esperar a que el resultado esté listo\n","    payload = {\n","        'CMD': 'Get',\n","        'RID': rid\n","    }\n","    while True:\n","        time.sleep(10)  # Esperar 10 segundos antes de verificar nuevamente\n","        response = requests.post(base_url, data=payload)\n","        response.raise_for_status()\n","        result_page = response.text\n","        if \"Status=WAITING\" in result_page:\n","            continue\n","        elif \"Status=READY\" in result_page:\n","            break\n","        else:\n","            raise Exception(\"Error en el estado de la búsqueda\")\n","\n","    # Extraer la secuencia alineada de la respuesta y guardarla en el archivo de salida\n","    result_page = result_page.split('\\n')\n","    alignment_started = False\n","    with open(output_file, 'w') as f:\n","        for line in result_page:\n","            if line.startswith('Sequences producing significant alignments:'):\n","                alignment_started = True\n","            elif alignment_started:\n","                if line.startswith('>'):\n","                    f.write('\\n' + line + '\\n')\n","                else:\n","                    f.write(line + '\\n')\n","    \n","    print('La secuencia alineada se ha guardado en el archivo', output_file)\n","#En esta versión de la función, después de esperar a que el resultado esté listo, la función extrae la secuencia alineada de la respuesta del servidor de BLAST y la guarda en el archivo de salida. La secuencia alineada se escribe en formato FASTA en el archivo, con una línea que comienza con el carácter '>' seguida del nombre de la secuencia y luego las líneas de la secuencia alineada.\n","#Para utilizar la función con tus archivos de entrada y salida, puedes llamarla de la siguiente manera:\n","\n","query_file = \"archivo_high.fasta\"\n","subject_file = \"archivo_unspecified.fasta\"\n","output_file = \"archivo_salida.blast\"\n","\n","run_blast(query_file, subject_file, output_file)"],"metadata":{"id":"zua219BUJfRn"},"id":"zua219BUJfRn","execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["# recomendaciones chatgpt-4"],"metadata":{"id":"85aaS5krBJZ6"},"id":"85aaS5krBJZ6"},{"cell_type":"code","source":["#Para BLAST, puedes usar la API de NCBI:\n","\n","import requests\n","from Bio.Blast import NCBIXML\n","\n","def run_blast(sequence, program='blastp', database='nr', email='your_email@example.com'):\n","    base_url = 'https://blast.ncbi.nlm.nih.gov/Blast.cgi'\n","    params = {\n","        'CMD': 'Put',\n","        'PROGRAM': program,\n","        'DATABASE': database,\n","        'QUERY': sequence,\n","    }\n","    response = requests.post(base_url, params=params)\n","    rid = response.text.split('&')[0].split('=')[1]\n","    \n","    params = {\n","        'CMD': 'Get',\n","        'FORMAT_TYPE': 'XML',\n","        'RID': rid,\n","    }\n","    \n","    while True:\n","        response = requests.get(base_url, params=params)\n","        if response.status_code == 200:\n","            break\n","    \n","    return NCBIXML.read(StringIO(response.text))\n","\n","\n","#Para Clustal Omega, puedes usar la API de EMBL-EBI:\n","import requests\n","\n","def run_clustal_omega(fasta_sequences, email='your_email@example.com'):\n","    base_url = 'https://www.ebi.ac.uk/Tools/services/rest/clustalo'\n","    headers = {'Content-Type': 'application/x-www-form-urlencoded'}\n","    \n","    params = {\n","        'email': email,\n","        'sequence': fasta_sequences,\n","    }\n","    \n","    response = requests.post(f'{base_url}/run', data=params, headers=headers)\n","    job_id = response.text\n","    \n","    while True:\n","        response = requests.get(f'{base_url}/status/{job_id}')\n","        if response.text == 'FINISHED':\n","            break\n","    \n","    response = requests.get(f'{base_url}/result/{job_id}/out')\n","    return response.text\n","\n","#Puedes utilizar la siguiente función para cargar archivos desde tu computadora en Google Colab:\n","from google.colab import files\n","uploaded = files.upload()\n","#Para descargar archivos generados en Google Colab, puedes usar:\n","from google.colab import files\n","files.download('path/to/your/file')"],"metadata":{"id":"vSASWwRDBCp2"},"id":"vSASWwRDBCp2","execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["# Salvar"],"metadata":{"id":"ZIpnsSMnOssQ"},"id":"ZIpnsSMnOssQ"},{"cell_type":"code","source":["#clustal = '\"clustal-omega-1.2.2-win64/clustalo.exe\"' + ' -i ' + archivo_entrada + ' -o ' + archivo_salida + ' --outfmt phylip -v --force'\n","clustal = 'clustalo' + ' -i ' + archivo_entrada + ' -o ' + archivo_salida + ' --outfmt phylip -v --force'\n","result = run(clustal, stdout=PIPE, shell=True, check=True)"],"metadata":{"id":"VTHlSJaXOqhb"},"id":"VTHlSJaXOqhb","execution_count":null,"outputs":[]},{"cell_type":"code","source":["ntiendo, para extraer solo la tercera línea que corresponde a la secuencia de la consulta, podemos usar expresiones regulares en Python. Aquí está el código:\n","\n","import re\n","\n","# Definir la expresión regular para encontrar la secuencia de la consulta:\n","pattern = re.compile(r'^query\\s+(\\d+)\\s+([ACGT-]+)$', re.MULTILINE)\n","\n","# Buscar la secuencia de la consulta en la cadena 'x':\n","match = pattern.search(x)\n","\n","# Obtener la secuencia de la consulta sin la palabra 'query' ni el número:\n","sequence = match.group(2).replace(' ', '').replace('-', '')\n","\n","# Imprimir la secuencia resultante:\n","print(sequence)\n","En este código, utilizamos una expresión regular que busca una línea que comience con la palabra 'query', seguida de un número y la secuencia de caracteres de '[ACGT-]'. Usamos el modificador re.MULTILINE para hacer que la expresión regular coincida con múltiples líneas en la cadena x. Luego, extraemos la secuencia de la consulta usando el método group(2) del objeto de coincidencia de la expresión regular, lo que nos proporciona la secuencia de caracteres de la tercera línea. Finalmente, eliminamos los espacios en blanco y los guiones de la secuencia resultante utilizando el método replace().\n","\n","Si ejecutas este código, debería imprimir solo la secuencia de la consulta sin la palabra 'query' ni el número."],"metadata":{"id":"GcvP8d04ZptV"},"id":"GcvP8d04ZptV","execution_count":null,"outputs":[]}],"metadata":{"colab":{"provenance":[]},"kernelspec":{"display_name":"Python 3","language":"python","name":"python3"},"language_info":{"codemirror_mode":{"name":"ipython","version":3},"file_extension":".py","mimetype":"text/x-python","name":"python","nbconvert_exporter":"python","pygments_lexer":"ipython3","version":"3.8.3"},"papermill":{"default_parameters":{},"duration":2023.773586,"end_time":"2023-01-08T13:54:56.309488","environment_variables":{},"exception":null,"input_path":"__notebook__.ipynb","output_path":"__notebook__.ipynb","parameters":{},"start_time":"2023-01-08T13:21:12.535902","version":"2.3.4"},"gpuClass":"standard"},"nbformat":4,"nbformat_minor":5}